src/celengine/astro.cpp

```diff
@@ -364,35 +419,35 @@ const char* astro::Date::toCStr(Format format) const
         break;
     }
 
-    strftime(date, sizeof(date), strftime_format, &cal_time);
+    auto length = std::strftime(result.data(), result.size(), strftime_format, &cal_time);
```

Comment:
could you check if we can use `fmt` here as well?

Reply:
Have put something in but am currently on Windows, so need to do some testing with this code path active.